src/ufuncs.c

```diff
@@ -1945,6 +1945,78 @@ static void polygonize_full_func(char** args, npy_intp* dimensions, npy_intp* st
 }
 static PyUFuncGenericFunction polygonize_full_funcs[1] = {&polygonize_full_func};
 
+static char nearest_points_dtypes[3] = {NPY_OBJECT, NPY_OBJECT, NPY_OBJECT};
```

Comment:
The "nearest_points" function here does not return points, but a line. So I think it better be named `shortest_line` in the ufuncs.c (just like in the python function).

Reply:
Indee, I didn't yet rename the actual ufunc after renaming the public function